File: problemset-new/004/00441-arranging-coins/freedomDR.cpp
#include <bits/stdc++.h>

using namespace std;
class Solution {
public:
    int arrangeCoins(int n) {
        int lo = 0, hi = 1<<16;
        while(lo < hi) {
            int mid = lo+(hi-lo)/2;
            int temp = 1LL*mid * ((mid+1)/2) + (mid%2 ? 0 : mid/2);
            if(temp < n) lo = mid+1;
            else hi = mid;
        }
        if(1LL*lo*((lo+1)/2)+(lo%2?0:lo/2) == n) return lo;
        else return lo-1;
    }
};
int main()
{
    return 0;
}

File: problemset-new/004/00441-arranging-coins/freedomDR.go
package main

import "fmt"

func arrangeCoins(n int) int {
	lo, hi := 0, int(1<<16)
	for lo < hi {
		mid := lo + (hi-lo)/2
		t := mid * ((mid + 1) / 2)
		if mid%2 == 0 {
			t += mid / 2
		}
		if t < n {
			lo = mid + 1
		} else {
			hi = mid
		}
	}
	t := lo * ((lo + 1) / 2)
	if lo%2 == 0 {
		t += lo / 2
	}
	if t == n {
		return lo
	} else {
		return lo - 1
	}
}
func main() {
	fmt.Println("vim-go")
}
